text/0002-bundle-add-and-remove.md

```diff
@@ -0,0 +1,275 @@
+
+## 1. adding gem to group block
+
+### 1.1. when Gemfile doesn't yet have the group specified
+
+#### Proposed command
+
+```ruby
+# Gemfile before
+source "https://rubygems.org"
+```
+
+```ruby
+bundle add minitest --group test
+```
+
+```ruby
+# Gemfile after
+source "https://rubygems.org"
+
+group :test do
+  gem "minitest", "~> 5.11"
+end
+```
+
+### 1.2. when Gemfile has the group specified
+
+#### Proposed command
+
+```ruby
+# Gemfile before
+source "https://rubygems.org"
+
+group :test do
+  gem "minitest", "~> 5.11"
+end
+```
+
+```shell
+bundle add minitest-reporters --group test
+```
+
+```ruby
+# Gemfile after
+source "https://rubygems.org"
+
+group :test do
+  gem "minitest", "~> 5.11"
+  gem "minitest-reporters" "~> 1.1"
+end
+```
+
+### Description
+
+Instead of adding gems to the Gemfile in the format
+
+```ruby
+gem "minitest", "~> 5.11", group: [:test]
+```
+
+the `add` command works with group blocks
+
+
+```ruby
+group :test do
+  gem "minitest", "~> 5.11"
+end
+```
+
+which are more natural for users.
+
+It appends gems to the end of the group block if the group already exists. It also supports adding to multiple groups.
+
+## 2. adding multiple gems with options
+
+#### Proposed command
+
+```shell
+bundle add sinatra:2.0 minitest:~>5.11
+```
+
+### Description
+
+The `add` command supports specifying versions for multiple gems.
+
+## 3. adding gems with optimistic or strict version ranges
+
+### 3.1 adding gems with strict version
+
+#### Proposed commands
+
+```shell
+bundle add sinatra minitest --strict
+```
+
+```ruby
+# Gemfile
+gem "sinatra", "2.0.0"
+gem "minitest", "5.11.3"
+```
+
+### 3.2 adding gems with optimistic version range
+
+#### Proposed commands
+
+```shell
+bundle add sinatra minitest --optimistic
+```
+
+```ruby
+# Gemfile
+gem "sinatra", ">= 2.0.0"
+gem "minitest", ">= 5.11.3"
+```
+
+### 3.3 adding gems without the version range
+
+```shell
+bundle add sinatra minitest --no-version
+```
+
+```ruby
+# Gemfile
+gem "sinatra"
+gem "minitest"
+```
+
+### Description
+
+The `add` command supports specifying different version ranges.
+
+## 4. skipping install step
+
+#### Proposed commands
+
+```shell
+bundle add sinatra minitest --skip-install
+```
+
+### Description
+
+The `--skip-install` flag skips the install step and just adds the gem to Gemfile.
+
+## 5. removing gems
+
+#### Proposed commands
+
+```ruby
+# Gemfile before
+source "https://rubygems.org"
+
+group :test do
+  gem "minitest"
+  gem "minitest-reporters"
+end
+```
+
+```shell
+bundle remove minitest minitest-reporters
+bundle rm minitest minitest-reporters
+```
+
+```ruby
+# Gemfile after
+source "https://rubygems.org"
+```
+
+### Description
+
+The `remove` command removes gems from the Gemfile, along with their corresponding group block if it remains empty after the deletion. By default it runs `bundle install` afterwards, which can also be skipped with the `--skip-install` flag.
```

Comment:
It is `remove` or `rm`? I'm in favor of `remove`, for symmetry with `add`

Reply:
I too am in favour of `remove` for same symmetry reason and also most bundler commands are whole words, not abbreviations/shortened versions. User will therefore expect whole words as commands.